x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/accesscontrol/OptOutQueryCache.java

```diff
@@ -50,6 +59,11 @@ public void clear(String reason) {
 
     @Override
     public Weight doCache(Weight weight, QueryCachingPolicy policy) {
+        if (licenseState.isAuthAllowed() == false) {
```

Comment:
Is it intentional that this is looking only at `isAuthAllowed`  and not `isSecurityEnabled` ?

The implementation here will disable the cache for trial licenses even if they have not enabled security.

Reply:
This should check `isSecurityEnabled`. The other methods like isAuthAllowed do not take into account the default distribution behavior where security is disabled with a new trial license unless it is explicitly enabled.

Reply:
I do not think this is correct. I think that `LicenseState#isSecurityEnabled` will be true for a basic license.

Reply:
I think that it should be `licenseState.isSecurityEnabled() == false || licenseState.isAuthAllowed() == false`

Reply:
@jaymode @tvernum I pushed 7bae967a2edbd245f5e9db9ea0a8634388aa70c4.